src/types/migration/ContentRelationship.ts

```diff
@@ -13,17 +15,16 @@ export type MigrationContentRelationship<
 	TDocuments extends PrismicDocument = PrismicDocument,
 > =
 	| ValueOrThunk<TDocuments | PrismicMigrationDocument<TDocuments> | undefined>
-	| (Pick<FilledContentRelationshipField, "link_type"> &
-			Partial<Pick<FilledContentRelationshipField, "text">> & {
-				id: ValueOrThunk<
-					TDocuments | PrismicMigrationDocument<TDocuments> | undefined
-				>
-			})
+	| (Pick<ContentRelationshipField, "link_type"> & {
```

Comment:
❓ Why did you remove text here? It's used in `resolveMigrationContentRelationship` and from what I understood with Lucie will still be used while handling link to document that can be from a generic link.

Reply:
`text` was removed because this type defines how users specify a content relationship value. `text` is invalid for content relationship fields.

Internally, we needed to widen the types *after* verifying the shape, not before. You can see the `& OptionalLinkProperties` in `src/lib/isValue.ts` and `src/lib/isMigrationValue.ts`.

These changes support the following behavior when migrating a document:
- Content relationship fields will not include `text` or `variant`.
- Link to media fields will not include variant. It includes `text` only because we explicitly added it to the field type. We can easily remove it for this field type only if we decide to do so.
- Link fields include `text` and `variant`.

Reply:
Hum, I'm not sure to get everything.

> Content relationship fields will not include text or variant.

That I understand, but what's the difference with link to document? What I understood from Lucie is that the type is used also for migrating link to document implying that text and variant are supported. 
There is no difference between link to document and content relationship behind the scene. Right?

> Link to media fields will not include variant. It includes text only because we explicitly added it to the field type. We can easily remove it for this field type only if we decide to do so.

For me, it's either both text + variant or nothing. We are waiting for the product but meanwhile, I would go for both. (details we can take care of course)

Reply:
> That I understand, but what's the difference with link to document? What I understood from Lucie is that the type is used also for migrating link to document implying that text and variant are supported.
There is no difference between link to document and content relationship behind the scene. Right?

Correct; content relationship functions are used for both content relationship fields and link fields. From the public API, we want to separate these field types so they only accept what is allowed. From the internal API, we need to treat both field types as overlapping.

Before this PR, we made the overlap public, which meant devs could assign `text` to a content relationship field, which should not be allowed.

Here is what the autocomplete looks like now for each field type:

| Link type | Autocomplete | Notes |
| --- | --- | --- |
| Link field | <img width="300" alt="image" src="https://github.com/user-attachments/assets/96e6c599-44ce-4cf4-aff2-be58aa03c68c" /> | Includes `text` and `variant`, along with the other migration-specific properties.
| Link to media | <img width="300" alt="image" src="https://github.com/user-attachments/assets/b87e7119-2daf-4b58-99ea-1a393fbe5554" /> | Includes `text`. |
| Content relationship | <img width="300" alt="image" src="https://github.com/user-attachments/assets/038af959-db2b-4d4e-ac3a-4b3a8cc55dac" /> | Does not include `text` or `variants`. |

> For me, it's either both text + variant or nothing. We are waiting for the product but meanwhile, I would go for both. (details we can take care of course)

I'm happy to remove `text` from link to media if we want to do that. I left it in since that's how the product works today. Are you good with removing it from `@prismicio/client` now?